compiler-core/src/dependency.rs

```diff
@@ -63,6 +63,61 @@ where
     Ok(packages)
 }
 
+/**
+* Used to compare 2 versions of a package.
+*/
+pub type PackageVersionDiffs = HashMap<String, (Version, Version)>;
```

Comment:
Let's either use this always or remove it

Reply:
> Let's either use this always or remove it

I'm not sure what you mean. `PackageVersionDiffs` are used only for the major versions check currently. I don't see any other place making comparisons between 2 versions (except probably within pubgrub which makes comparisons betweens many versions probably).

Is you suggestion to remove the type and use `HashMap<String, (Version, Version)>` everywhere?